### PR TITLE
LPD-21348 Update ObjectDefinition userId after user being deleted.

### PR DESCRIPTION
**Pull request by thallescezar**
Related issue: https://liferay.atlassian.net/browse/LPD-21348

### CODE REVIEW

**Review by linolaoj — APPROVED** [review]

**Comment by linolaoj** [comment]
ci:forward

**Comment by linolaoj** [comment]
Hi @thallescezar, [SemanticVersioningTest](https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/881639/testReport/com.liferay.semantic.versioning/SemanticVersioningTest) is related, we need to run the baseline task.

**Comment by linolaoj** [comment]
ci:test:relevant

**Comment by linolaoj** [comment]
Hi @thallescezar can you please rebase and resend, you might need to run the baseline again to rebase.

**Comment by linolaoj** [comment]
Hi @thallescezar , can you please check if the test `com.liferay.object.storage.salesforce.internal.rest.manager.v1_0.test.SalesforceObjectEntryManagerImplTest.testGetObjectEntries` is related? If it is not we can forward force it.

**Comment by thallescezar** [comment]
Going to check on it, @linolaoj

**Comment by thallescezar** [comment]
ci:forward:force

**Review by linolaoj — APPROVED** [review]

**Comment by linolaoj** [comment]
ci:forward

**Comment by linolaoj** [comment]
ci:test:stable

**Comment by linolaoj** [comment]
Hi @thallescezar, we will need to add the upgrade step test. 
https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Testing

**Comment by linolaoj** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/model/listener/test/UserModelListenerTest.java [hunk 1]
```diff
@@ -91,6 +96,34 @@ public void tearDown() throws Exception {
...
+			_objectDefinitionLocalService.createObjectDefinition(
+				RandomTestUtil.randomLong());
+
+		User user = UserTestUtil.addUser();
+
+		objectDefinition.setCompanyId(user.getCompanyId());
+		objectDefinition.setUserId(user.getUserId());
+
+		objectDefinition = _objectDefinitionLocalService.updateObjectDefinition(
+			objectDefinition);
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/model/listener/test/UserModelListenerTest.java:111]: Use `_objectDefinitionLocalService.addCustomObjectDefinition` instead. Like https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java#L1836-L1843

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		ObjectDefinition objectDefinition = _addCustomObjectDefinition(
+			"A" + RandomTestUtil.randomString(),
+			"A" + RandomTestUtil.randomString(),
+			"A" + RandomTestUtil.randomString(), userId);
+
+		return _objectDefinitionLocalService.publishCustomObjectDefinition(
+			userId, objectDefinition.getObjectDefinitionId());
+	}
+
+	private void _runUpgrade() throws Exception {
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java:128]: Method `_runUpgrade` not needed. Inline it

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+		defaultServiceAccountUserId = _userLocalService.getUserIdByScreenName(
+			company2.getCompanyId(), "default-service-account");
+
+		_assertObjectDefinitionUserId(
+			defaultServiceAccountUserId,
+			objectDefinition2.getObjectDefinitionId());
+	}
+
+	private ObjectDefinition _addCustomObjectDefinition(
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java:88]: Method `_addCustomObjectDefinition` not needed. Inline it.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _CLASS_NAME);
+
+			upgradeProcess.upgrade();
+		}
+	}
+
+	private static final String _CLASS_NAME =
+		"com.liferay.object.internal.upgrade.v9_1_2." +
+			"ObjectRelationshipUpgradeProcess";
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java:141]: Remember to rename the log when renaming the upgrade file

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionId);
+
+		Assert.assertEquals(expectedUserId, objectDefinition.getUserId());
+	}
+
+	private ObjectDefinition _publishCustomObjectDefinition(long userId)
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java:116]: Better to inline the method `_publishCustomObjectDefinition` also to follow the pattern we have in `ObjectDefinitionLocalServiceTest`

**Review by gabrielwas — CHANGES_REQUESTED** [review]
@thallescezar @linolaoj check my suggestions. Thx.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/UserModelListener.java [hunk 6]
```diff
@@ -29,6 +31,30 @@
 @Component(service = ModelListener.class)
 public class UserModelListener extends BaseModelListener<User> {
 
+	@Override
+	public void onAfterRemove(User user) throws ModelListenerException {
+		if (Objects.equals(user.getScreenName(), "default-service-account")) {
```
[the remark this answers is not in the document]
Comment (thallescezar) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/UserModelListener.java:36]: Tested and worked fine. Removing it

**Comment by linolaoj** [comment]
Hi, I'll rebase and resend with some changes for the test to compile.

**Comment by linolaoj** [comment]
@rijema , I sent you a pr with a wip of my changes, can you please check it and send a new pr?

**Pull request closed without merging**